exporter/exporterhelper/README.md

```diff
@@ -1,28 +1,99 @@
 # Exporter Helper
 
-This is a helper exporter that other exporters can depend on. Today, it
-primarily offers queued retries  and resource attributes to metric labels conversion.
+This is a helper exporter that other exporters can depend on. It offers several
+independent capabilities including:
+
+- Export settings
+- In-memory queuing
+- Ability to retry an export operation
 
 > :warning: This exporter should not be added to a service pipeline.
 
 ## Configuration
 
-The following configuration options can be modified:
+The below configuration options can be modified. The full list of settings
+exposed for this helper exporter are documented [here](factory.go).
+
+### Export Settings
+
+These settings modify the behavior when a payload is exported.
+
+- `resource_to_telemetry_conversion`
+  - `enabled` (default = `false`): If `enabled` is `true`, all the resource
+    attributes will be converted to metric labels.
+- `timeout` (default = `5s`): Time to wait per individual attempt to export data
+  to a backend. If the backend accepts the data or returns an error before the
+  configured `timeout` then the `timeout` setting is ignored. This setting is
+  completely independent of `max_elapsed_time` under `retry_on_failure.`
+
+### Queuing
+
+These settings modify in-memory queuing. It is recommended to keep queuing
+enabled so data is not lost during export in case of error or timeout.
 
-- `retry_on_failure`
-  - `enabled` (default = true)
-  - `initial_interval` (default = 5s): Time to wait after the first failure before retrying; ignored if `enabled` is `false`
-  - `max_interval` (default = 30s): Is the upper bound on backoff; ignored if `enabled` is `false`
-  - `max_elapsed_time` (default = 120s): Is the maximum amount of time spent trying to send a batch; ignored if `enabled` is `false`
 - `sending_queue`
-  - `enabled` (default = true)
-  - `num_consumers` (default = 10): Number of consumers that dequeue batches; ignored if `enabled` is `false`
-  - `queue_size` (default = 5000): Maximum number of batches kept in memory before data; ignored if `enabled` is `false`;
+  - `enabled` (default = `true`)
+  - `num_consumers` (default = `10`): Number of consumers that dequeue batches;
```

Comment:
Is there a recommendation to get to the right value? Like, one consumer per 50 items in the queue_size? One consumer per CPU (I don't think so, as those are Go channels likely)? What kind of load can 10 consumers handle?

Reply:
leave the default :) same applies to other settings like `queue_size` -- how big should it be?

Reply:
Queue size can be kinds calculated by users: if I know the average size of my batch, I can calculate the queue size based on the amount of memory I want to use. For this one (num_consumers), it's kinda hard to figure out what's a good size for my setup. This is what we have for Jaeger, for a property with similar semantics: https://www.jaegertracing.io/docs/1.21/performance-tuning/#adjust-processor-workers-1

Reply:
Anyway, I think that it is not related to the issue actually. In my opinion, this is not obvious thing, deserving a separate issue